### BRAF-RAS Score (BRS): Information:
# This script reads in DESeq2 normalized expression data for our thyroid cohort and a 71 gene list for BRAF-RAS Score (BRS) and uses the gene list to create a BRS for each sample (HT, MNG excluded).
# The gene list for the BRS was generated by the 2014 PTC TCGA paper
# In brief, I will make BRAF mutant centroids and RAS mutant centroids for these gene lists. 
# The BRS for each individual sample will be based on the normalized squared euclidean distance from these centroids. 

#### library(tidyverse) ####

######## Step 1: Establish BRAF and RAS mutant patient sample lists ###########
# Read in clinical data and mutation data
ClinicalData <- read_csv(file = "data_in_use/Clindata_Complete.csv")
MutationData <- read_csv(file = "data_in_use/VUMC.cohort.Mutation.Data.csv")

# Subset Mutation data to be just columns of interest
MutationData <- MutationData[c("IP", "RNA.ID", "Diagnosis", "Diagnosis.with.HTPTC", "BRAF.mutation", "RAS.family.mutation")]

# Merge clinical data and Mutation Data
MutationData <- MutationData %>% as_tibble %>% merge(ClinicalData, by.x = "IP", by.y = "IP", all = T)

# Restrict to diagnosis = PTC or diagnosis = FVPTC and exclude HTPTC for centroid calculations
MutationData_PTCs <- MutationData %>% subset((Diagnosis == "PTC" | Diagnosis == "FVPTC") & Diagnosis.with.HTPTC != "HTPTC")

# Restrict MutationData_PTCs to NonMetastatic
MutationData_PTCs <- MutationData_PTCs %>% subset(Location.Type == "NonMet")

# Subset out just the BRAF mutants from mutation data PTCs
# Note, we now have 39 BRAF mutants (vs 37 prior). This is two additional data points for calculating scores. 
BRAF_Mutant_PTCs <- MutationData_PTCs %>% subset(BRAF.mutation == 1)

# Subset out just the RAS mutants from mutation data PTCs
RAS_Mutant_PTCs <- MutationData_PTCs %>% subset(RAS.family.mutation == 1)

# 22-0322 Note: I double checked and there is NO overlap in these lists (i.e., no samples with both BRAF and RAS mutation)

###### Step 2: Establish Gene list #######
#Read in BRAF-RAS Score Gene List
BRS_Genes <- read_csv(file = "data_in_use/21-1212_BRS_Gene_List_71.csv")

# Went through the gene list and FAM176A was not in the data set
# Per genecards.org, FAM176A is also EVA1A (https://www.genecards.org/cgi-bin/carddisp.pl?gene=EVA1A)
# EVA1A is in the data set...with the following for loop I will change FAM176A to EVA1A
# Similarly, per genecards.org https://www.genecards.org/cgi-bin/carddisp.pl?gene=DCSTAMP), TM7SF4 is DCSTAMP.
# I will also change TM7SF4 to DCSTAMP with this for loop.
# Note that there should be a more efficient way to do this
# Although PVRL4 is used by the TCGA data, NECTIN4 is the current name and what is used in our data. 
# I will change NECTIN4 here as well. 
for(i in 1:nrow(BRS_Genes)){
  if(BRS_Genes$BRS_Gene_list[i] == "FAM176A"){
    BRS_Genes$BRS_Gene_list[i] <- "EVA1A"
  }
  if(BRS_Genes$BRS_Gene_list[i] == "TM7SF4"){
    BRS_Genes$BRS_Gene_list[i] <- "DCSTAMP"
  }
  if(BRS_Genes$BRS_Gene_list[i] == "PVRL4"){
    BRS_Genes$BRS_Gene_list[i] <- "NECTIN4"
  }
}

# Note: also found no match for FLJ23867 (ncRNA/uncharacterized protein https://www.ncbi.nlm.nih.gov/gene/200058)
# Removing from gene list -> list now length 70
BRS_Genes <- filter(BRS_Genes, BRS_Gene_list != "FLJ23867")

# Removing ANXA2P2. It is in the TCGA data. However, it is not in our data set.
# List now length 69. 
BRS_Genes <- filter(BRS_Genes, BRS_Gene_list != "ANXA2P2")

# Put the gene list in alphabetical order
BRS_Genes <- BRS_Genes %>% arrange(BRS_Gene_list)


########## Step 3: Formatting Internal DESeq2 centroids to calculate Z-Scores ###########
Thyroid_Cohort_RNA <- read.table(file = "data_in_use/DESeq2_Z-Scores.txt", sep = '\t', header = TRUE)
Thyroid_Cohort_RNA <- as_tibble(Thyroid_Cohort_RNA) # Change type to tibble

# Restrict data to just RNA IDs and the genes in the gene list
Thyroid_Cohort_RNA_GeneSet <- Thyroid_Cohort_RNA[c("RNA.ID",BRS_Genes$BRS_Gene_list)]

# Remove 'X' from the beginning of Thyroid_Cohort_RNA_GeneSet column RNA.ID
Thyroid_Cohort_RNA_GeneSet$RNA.ID <- sub('.','', Thyroid_Cohort_RNA_GeneSet$RNA.ID)

# Restrict sequencing data for BRAF and RAS cohorts for calculating centroids and remove RNA.ID column
BRAF_Thyroid_Cohort <- Thyroid_Cohort_RNA_GeneSet %>% filter(RNA.ID %in% c(BRAF_Mutant_PTCs$RNA.ID)) # Only include RNA IDs that are BRAF mutants
BRAF_Thyroid_Cohort <- BRAF_Thyroid_Cohort[c(BRS_Genes$BRS_Gene_list)] # Remove RNA ID column
RAS_Thyroid_Cohort <- Thyroid_Cohort_RNA_GeneSet %>% filter(RNA.ID %in% c(RAS_Mutant_PTCs$RNA.ID)) # Only include RNA IDs that are RAS mutants
RAS_Thyroid_Cohort <- RAS_Thyroid_Cohort[c(BRS_Genes$BRS_Gene_list)] # Remove RNA ID column

# Calculate centroids
BRAF_Centroid <- apply(BRAF_Thyroid_Cohort, 2, median) # Note using median instead of mean here
RAS_Centroid <- apply(RAS_Thyroid_Cohort, 2, median) # Note using median instead of mean here

############ Step 4: Format thyroid data for normalized euclidean distances ############
# Create a header for future column names
Sequencing.ID <- Thyroid_Cohort_RNA_GeneSet$RNA.ID

# Remove the RNA ID header
Thyroid_Cohort_RNA_GeneSet <- Thyroid_Cohort_RNA_GeneSet[c(BRS_Genes$BRS_Gene_list)]

# Transpose the matrix to put it in the appropriate order for calculating euclidean distances
Thyroid_Cohort_RNA_GeneSet <- t(Thyroid_Cohort_RNA_GeneSet)

# Add the sample IDs back on top as column names
colnames(Thyroid_Cohort_RNA_GeneSet) <- Sequencing.ID
rownames(Thyroid_Cohort_RNA_GeneSet) <- NULL #remove row names

############## Step 5: Calculating Normalized Euclidean Distance ########################
## Define function for Euclidean Distance
# Adapt the normalized euclidean formula for vectors of length n using a for loop
Normalized_Euclidean_For_Loop <- function(x,y){
  numerator_count <- 0
  denomenator_count <- 0
  for(i in 1:length(x)){
    numerator_count = numerator_count + 
      (abs(x[i]+(1/length(x))*(-sum(x))-y[i]+(sum(y))/length(y))^2)
    denomenator_count <- denomenator_count + 
      abs(x[i]+(1/length(x))*(-sum(x)))^2+
      abs(y[i]+(1/length(y))*(-sum(y)))^2
  }
  denomenator_count = denomenator_count*2
  return(numerator_count/denomenator_count)
}

# Generate BRS_Scores, a vector that is the length of our cohort (299 samples) and will hold the BRS score for each sample
Internal_BRS_Scores <- numeric(length = ncol(Thyroid_Cohort_RNA_GeneSet))

# Fill in BRS_Scores with the BRS Score for each sample by calling the normalized euclidean distance algorithm for each sample 
# Call the algorithm for each sample compared to the RAS centroid and the BRAF centroid. 
# Subtract the RAS value from the BRAF value
for (i in 1:ncol(Thyroid_Cohort_RNA_GeneSet)){
  RAS_Holder <- Normalized_Euclidean_For_Loop(Thyroid_Cohort_RNA_GeneSet[,i], RAS_Centroid)
  BRAF_Holder <- Normalized_Euclidean_For_Loop(Thyroid_Cohort_RNA_GeneSet[,i], BRAF_Centroid)
  Internal_BRS_Scores[i] <- BRAF_Holder - RAS_Holder
}

Internal_BRS_Scores_Output <- data.frame(Sequencing.ID, Internal_BRS_Scores)
write.csv(Internal_BRS_Scores_Output, "data_in_use/22-0322_Internal-Z_BRS_Scores_Median.csv", row.names = FALSE)
